USER_GUIDE.md

```diff
@@ -40,44 +40,43 @@ var client = new Client({
 ### To authenticate with [Amazon OpenSearch Service](https://aws.amazon.com/opensearch-service/) using AwsSigv4Signer
 
 ```javascript
-const { defaultProvider } = require("@aws-sdk/credential-provider-node");
+const AWS = require('aws-sdk'); // V2 SDK.
+const { defaultProvider } = require("@aws-sdk/credential-provider-node"); // V3 SDK.
 const { Client } = require('@opensearch-project/opensearch');
 const { AwsSigv4Signer } = require('@opensearch-project/opensearch/aws');
 
-const endpoint = ""; // OpenSearch domain URL e.g. https://search-xxx.region.es.amazonaws.com
-
-async function getClient() {
-  const connection = await AwsSigv4Signer({
+const client = new Client({
+  ...AwsSigv4Signer({
     region: 'us-east-1',
-    // Must return an AWS.Credentials object.
-    // This function is used when initializing the client and
+    // Must return a Promise that resolve to an AWS.Credentials object.
+    // This function is used to acquire the credentials when the client start and
     // when the credentials are expired.
-    // Example with aws sdk V3:
+    // The Client will refresh the Credentials only when they are expired.
+    // With AWS SDK V2, Credentials.refreshPromise is used when available to refresh the credentials.
+
+    // Example with AWS SDK V3:
     getCredentials: async () => {
```

Comment:
Don't think we need `async` here. Could be confusing.

Reply:
right, let's get rid of it, kept it to make it explicit that this function can be asynchronous.